### PR TITLE
#2275. Add more noSuchMethod tests. Part 1

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks good, just one typo.

However, we might need to do something about external declarations and `dart2js`.

File: Language/Classes/Instance_Methods/Method_noSuchMethod/definition_A01_t03.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+///
+/// @description Checks that it is possible to define a `noSuchMethod` which
+/// is a correct override of `noSuchMethod` in `Object`.
+/// @author sgrekhov22@gmail.com
+
+import "../../../../Utils/expect.dart";
+
+class C1 {
+  int m();
+  external dynamic noSuchMethod(Invocation i);
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply Language/Classes/Instance_Methods/Method_noSuchMethod/definition_A01_t03.dart:30]: Yes, we turn-off these tests in `.status` files with the comment `SkipByDesign # Non-JS-interop external members are not supported`

**Comment by sgrekhov** [comment]
Typos fixed. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**